content/blog/duckplyr-1-0-0/index.Rmd

```diff
@@ -0,0 +1,286 @@
+This is a new dplyr backend powered by [DuckDB](https://duckdb.org/), a fast in-memory analytical database system[^duckdb].
+It joins the rank of dplyr backends together with [dtplyr](https://dtplyr.tidyverse.org) and [dbplyr](https://dbplyr.tidyverse.org).
+You can use it instead of dplyr for data small or large.
+
+<!-- FIXME:
+
+We have many more dplyr backends, the two above are just from the tidyverse.
+GitHub search: https://github.com/search?q=org%3Acran+%2FS3method%5B%28%5D%28mutate%7Csummarise%29+*%2C%2F&type=code
+Do we need an "awesome dplyr" like https://github.com/krlmlr/awesome-vctrs/?
+
+-->
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("duckplyr")
+```
+
+This article shows how duckplyr can be used instead of dplyr with data of different size, explain how you can help improve the package, and share a selection of further resources.
+
+## A drop-in replacement for dplyr
+
+Imagine you have to wrangle a huge dataset.
+Here we generate one using the [data generator from the TPC-H benchmark](https://duckdb.org/2024/04/02/duckplyr.html#benchmark-tpc-h-q1).
+
+```{r}
+lineitem <- duckdb:::sql("INSTALL tpch; LOAD tpch; CALL dbgen(sf=1); FROM lineitem;")
+lineitem <- tibble::as_tibble(lineitem)
+dplyr::glimpse(lineitem)
+```
+
+We could transform the data using dplyr but we could also transform it using a tool that'll scale well to ever larger data: duckplyr.
+The duckplyr package is a _drop-in replacement for dplyr_ that uses _DuckDB for speed_.
+You can simply _drop_ duckplyr into your pipeline by loading it, then computations will be efficiently carried out by DuckDB.
+
+[^duckdb]: If you haven't heard about it, you can watch [Hannes Mühleisen's keynote at posit::conf(2024)](https://www.youtube.com/watch?v=GELhdezYmP0&feature=youtu.be).
+
+Below, we express the standard "TPC-H benchmark query 1" in dplyr syntax, but execute it with duckplyr.
+
+```{r}
+library(conflicted)
+library(duckplyr)
+conflict_prefer("filter", "dplyr", quiet = TRUE)
+out <- lineitem |>
+  select(l_shipdate, l_returnflag, l_linestatus, l_quantity, l_extendedprice, l_discount, l_tax) |>
+  filter(l_shipdate <= !!as.Date("1998-09-02")) |>
+  select(l_returnflag, l_linestatus, l_quantity, l_extendedprice, l_discount, l_tax) |>
+  summarise(
+    sum_qty = sum(l_quantity),
+    sum_base_price = sum(l_extendedprice),
+    sum_disc_price = sum(l_extendedprice * (1 - l_discount)),
+    sum_charge = sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)),
+    avg_qty = mean(l_quantity),
+    avg_price = mean(l_extendedprice),
+    avg_disc = mean(l_discount),
+    count_order = n(),
+    .by = c(l_returnflag, l_linestatus)
+  ) |>
+  arrange(l_returnflag, l_linestatus)
+out
+```
+
+Like with other dplyr backends like dtplyr and dbplyr, duckplyr allows you to get faster results without learning a different syntax.
+Unlike other dplyr backends, duckplyr does not require you to change existing code or learn specific idiosyncracies.
```

Comment:
I'm wondering whether this sentence is slightly misleading: to use duckplyr efficiently, it's probably best to read the "limits" vignette, which is akin to learning about idiosyncrasies?